drivers/sensor/bh1749/bh1749.c

```diff
@@ -0,0 +1,287 @@
+				   BH1749_SYSTEM_CONTROL,
+				   BH1749_SYSTEM_CONTROL_SW_RESET_Msk,
+				   BH1749_SYSTEM_CONTROL_SW_RESET);
+}
+
+static int bh1749_rgb_gain_set(struct device *dev, u8_t value)
+{
+	return i2c_reg_update_byte(dev, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				   BH1749_MODE_CONTROL1,
+				   BH1749_MODE_CONTROL1_RGB_GAIN_Msk,
+				   value);
+}
+
+static int bh1749_rgb_measurement_enable(struct device *dev, bool enable)
+{
+	u8_t en = enable ?
+		  BH1749_MODE_CONTROL2_RGB_EN_ENABLE :
+		  BH1749_MODE_CONTROL2_RGB_EN_DISABLE;
+
+	return i2c_reg_update_byte(dev, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				   BH1749_MODE_CONTROL2,
+				   BH1749_MODE_CONTROL2_RGB_EN_Msk,
+				   en);
+}
+
+static int bh1749_init(struct device *dev)
+{
+	int err;
+	struct bh1749_data *data = dev->driver_data;
+
+	/* Activation time 2ms */
+	k_sleep(2);
+
+	data->i2c = device_get_binding(DT_ROHM_BH1749_0_BUS_NAME);
+
+	if (data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!",
+			DT_ROHM_BH1749_0_BUS_NAME);
+		return -EINVAL;
+	}
+
+	(void)memset(data->sample_rgb_ir, 0, sizeof(data->sample_rgb_ir));
+	data->pdata = 0U;
+
+	err = bh1749_sw_reset(data->i2c);
+	if (err) {
+		LOG_ERR("Could not apply software reset.");
+		return -EIO;
+	}
+
+	err = bh1749_check(dev);
+	if (err) {
+		LOG_ERR("Communication with BH1749 failed with error %d", err);
+		return -EIO;
+	}
+
+	err = bh1749_rgb_measurement_enable(data->i2c, true);
+	if (err) {
+		LOG_ERR("Could not set measurement mode.");
+		return -EIO;
+	}
+
+
+	err = bh1749_rgb_gain_set(data->i2c, BH1749_MODE_CONTROL1_RGB_GAIN_1X);
+	if (err) {
+		LOG_ERR("Could not set RGB gain.");
+		return -EIO;
+	}
+
+
+#ifdef CONFIG_BH1749_TRIGGER
+	err = bh1749_gpio_interrupt_init(dev);
+	if (err) {
+		LOG_ERR("Failed to initialize interrupt with error %d", err);
+		return -EIO;
+	}
+
+	LOG_DBG("GPIO Sense Interrupts initialized");
+#endif /* CONFIG_BH1749_TRIGGER */
+
+
+	/* Interrupt line goes active after every measurement */
+	if (i2c_reg_update_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				BH1749_PERSISTENCE,
+				BH1749_PERSISTENCE_PERSISTENCE_Msk,
+				BH1749_PERSISTENCE_PERSISTENCE_8_SAMPLES)) {
+		LOG_ERR("Could not set persistence configuration.");
+		return -EIO;
+	}
+
+	if (i2c_reg_update_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				BH1749_MODE_CONTROL1,
+				BH1749_MODE_CONTROL1_MEAS_MODE_Msk,
+				BH1749_MODE_CONTROL1_MEAS_MODE_120MS)) {
+		LOG_ERR("Could not set measurement mode configuration.");
+		return -EIO;
+	}
+
+	LOG_DBG("Persistence set");
+
+	return 0;
+}
+
+static const struct sensor_driver_api bh1749_driver_api = {
+	.sample_fetch = &bh1749_sample_fetch,
+	.channel_get = &bh1749_channel_get,
+#ifdef CONFIG_BH1749_TRIGGER
+	.attr_set = &bh1749_attr_set,
+	.trigger_set = &bh1749_trigger_set,
+#endif
+};
+
+static struct bh1749_data bh1749_data;
+
+DEVICE_DEFINE(bh1749, DT_ROHM_BH1749_0_LABEL, bh1749_init,
+	      NULL, &bh1749_data, NULL,
```

Comment:
When a device does not implement the PM operations, it should use the "empty" control function (`device_pm_control_nop`), not NULL.